networkx/algorithms/tests/test_matching.py

```diff
@@ -408,6 +408,10 @@ def test_invalid(self):
         G = nx.path_graph(4)
         assert not nx.is_matching(G, {(0, 1), (1, 2), (2, 3)})
 
+    def test_invalid_edge(self):
+        G = nx.path_graph(4)
+        assert not nx.is_matching(G, {(0, 3), (1, 2)})
+
```

Comment:
```suggestion
        assert not nx.is_matching(G, {(0, 3), (1, 2)})
        assert not nx.is_matching(G, {(0, 4)}
        G = nx.path_graph(4, create_using=nx.DiGraph)
        assert nx.is_matching(G, {(0, 1)})
        assert not nx.is_matching(G, {(1, 0)})
```
These are some more tests to make sure all(?) types of invalid edges are flagged.